06_puppy/mohitnag/memstore.go

```diff
@@ -0,0 +1,38 @@
+package main
+
+import "fmt"
+
+// CreatePuppy creates a Puppy in memstore
+func (m *MemStore) CreatePuppy(p Puppy) error {
+	if _, ok := (*m)[p.ID]; !ok {
```

Comment:
MemStore is a map, so it already is an implicit pointer.
I think you should be able to write all of these functions as:
```go
func (m MemStore) CreatePuppy(p Puppy) error {
```
etc. (notice the value receiver instead of the pointer receiver in the method?
I'm not 100% sure if this still satisfies the interface though - gotta try ;)
If it does, then you can get rid of all the `(*m)[id]` and us `m[id]` instead which to me is more readable.

Reply:
Yes. No need of pointer receiver. Removed from everywhere